### PR TITLE
change ir attributes to a map

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix the issue mentioned here: https://github.com/sql-machine-learning/sqlflow/pull/837#discussion_r325839271

### CODE REVIEW

File: sql/ir_generator_test.go [hunk 1]
```diff
@@ -46,13 +46,13 @@ INTO mymodel;`
 	a.Equal("SELECT c1, c2, c3, c4\nFROM my_table", trainIR.Select)
 
 	// TODO(typhoonzero): should find out why the order of parsed attributes changes randomly
```
Comment (tonyyang-svail) [inline sql/ir_generator_test.go]: Please remove this out-dated TODO.

File: sql/ir_generator.go [hunk 2]
```diff
@@ -72,17 +72,17 @@ func generateTrainIR(slct *extendedSelect, connStr string) (*codegen.TrainIR, er
...
 		resolvedExpr, err := parseExpression(v)
 		if err != nil {
 			return nil, err
 		}
-		ret = append(ret, &codegen.Attribute{
-			Key:   k,
-			Value: resolvedExpr,
-		})
+		if _, ok := ret[k]; ok {
+			return nil, fmt.Errorf("duplicate attribute: %v=%v", k, resolvedExpr)
```
Comment (tonyyang-svail) [inline sql/ir_generator.go:83]: The parser already checks the attribute duplication at 

https://github.com/sql-machine-learning/sqlflow/blob/01730675a6505d1384393a04f647f2dc380a211e/sql/sql.y#L103-L111

It is safe to remove this check.

**Review by tonyyang-svail — DISMISSED** [review]
LGTM.

File: sql/ir_generator_test.go (hunk 1 above)
Comment (typhoonzero) [reply sql/ir_generator_test.go]: Done

File: sql/ir_generator.go (hunk 2 above)
Comment (typhoonzero) [reply sql/ir_generator.go:83]: Done

**Review by tonyyang-svail — APPROVED** [review]

File: sql/ir_generator.go (hunk 2 above)
Comment (tonyyang-svail) [reply sql/ir_generator.go:83]: Sorry, I meant to delete the check at `ir_generator.go` and keeps the check at `sql.y`.

**Pull request merged**